src/test/kotlin/com/igorwojda/string/longestword/challenge.kt

```diff
@@ -4,7 +4,12 @@ import org.amshove.kluent.shouldBeEqualTo
 import org.junit.jupiter.api.Test
 
 private fun longestWord(str: String): String {
-    TODO("not implemented")
+    var longest = ""
+    for (string in str.split("\\W".toRegex()))
+        if (string.length > longest.length) {
```

Comment:
kotlin has some interesing aggregate operations you might want to use insetad: https://kotlinlang.org/docs/reference/collection-aggregate.html